src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -304,14 +296,13 @@ public void AddRange(T[] items, int length)
                 Requires.NotNull(items, "items");
                 Requires.Range(length >= 0, "length");
 
+                if (length > items.Length)
+                    throw new IndexOutOfRangeException("length");
```

Comment:
I agree with @AArnott that people should not be catching this anyway. Assuming failures Requires.Range checks throw and ArgumentOutOfRangeException then I'm OK with this change.